### PR TITLE
WIP:BUG: Remote modules Python files were built in the wrong folder

### PR DESCRIPTION
**Pull request by fbudin69500**
Remote modules Python files (*.so, *.py) were built in the remote module folder
instead of being build in ITK build folder. The problem has been introduced
in a previous commit [1]. This was not allowing to use ITK
in Python by simplying adding its build folder and subfolders to the Python
environment (using WrapITK.pth).
As a side effect, `WrapITKConfig.cmake` is removed as it was not used
anywhere. This file was not up to date and did not contain any useful
information except for the pixel types that are wrapped when buidling
ITK. The pixel type information has been moved to ITKConfig.cmake
which now contains all the useful information when importing ITK in
another project.

[1] b45161f558ec47f8233991cbad14b3b070836018

### CODE REVIEW

**Comment by fbudin69500** [comment]
@LucasGandel This is a follow up of a patch you submitted last year. I am not sure what was your exact goal. Could you explain what you tried to correct in your patch? https://github.com/InsightSoftwareConsortium/ITK/commit/b45161f558ec47f8233991cbad14b3b070836018

**Comment by fbudin69500** [comment]
@LucasGandel Thanks for the comment. That helps me to understand the purpose of your previous patch. I think that this patch should keep things working for you (give you access to the `ITK_WRAP_*) variables. I am still testing it locally. If you have something in mind that I should test it against, please let me know, so I don't break anything for you.

File: CMake/ITKConfig.cmake.in [hunk 1]
```diff
@@ -118,4 +123,38 @@ if(NOT DEFINED ITK_WRAP_TCL)
   set(ITK_WRAP_TCL "@ITK_WRAP_TCL@")
 endif()
 
+# Set up the rest of the variables that WrapITK was built with.
+set(ITK_WRAP_unsigned_char @ITK_WRAP_unsigned_char@ CACHE BOOL "Wrap unsigned char type")
```
Comment (thewtex) [inline CMake/ITKConfig.cmake.in]: If these are in `ITKConfig.cmake`, we do not want use `CACHE BOOL`.

**Comment by fbudin69500** [comment]
@thewtex We should merge this PR too if possible for ITK5 since it allows to use remote modules with Python ON and use them in the ITK build tree. This is an important bug to correct in my opinion.

**Pull request merged**